### ISSUE TITLE
Lint to prevent people from indexing a string via iterator.nth(i)

### ISSUE THREAD

**Issue by sebschmi** [rust-lang/rust-clippy#6391]
### What it does

It is possible to access the nth byte of a string using `string.bytes().nth(i)`. But in case of `i = 0` this collides with the lint iter-nth-zero.
Alternatively one could write `string.as_bytes().get(1)`, which does not produce any clippy warnings. This code is equivalent to the code above in what it does.

### Categories (optional)

- Kind: maybe `clippy::style` and maybe `clippy::pedantic`. Maybe also `clippy::perf`.

### Advantages 

- Does not wrongly warn about iter-nth-zero when accessing the first byte of a string
- Does not rely on the compiler to optimise out the iteration when doing direct indexed access

### Drawbacks

The iteration is very easy for the compiler to optimise out, so that argument is weak.
The argument about iter-nth-zero is valid but it might be better to patch iter-nth-zero instead of teaching people how to code around the drawbacks of iter-nth-zero.

Maybe iter-nth-zero should become pedantic instead?

### Example

```rust
string.bytes().nth(i)
```

Could be written as:

```rust
string.as_bytes().get(i)
```

**Comment by TaKO8Ki** [comment]
Can I tackle this issue?

**Comment by giraffate** [comment]
I think you can take this.

**Comment by TaKO8Ki** [comment]
I'm going to tackle this issue soon.
